src/map.cpp

```diff
@@ -4949,6 +4950,13 @@ std::list<item> map::use_charges( const tripoint &origin, const int range,
         }
     }
 
+    if( bcp ) {
+        quantity = bcp->use_charges( type, quantity, ret );
+        if( quantity == 0 ) {
```

Comment:
Probably better to make this `<=` like other places in this function for safety if something goes wrong.
```suggestion
        if( quantity <= 0 ) {
```